src/ethereum_test_tools/rpc/rpc.py

```diff
@@ -0,0 +1,118 @@
+"""
+JSON-RPC methods and helper functions for EEST consume based hive simulators.
+"""
+
+import time
+from typing import Dict, List, Literal, Union
+
+import requests
+from jwt import encode
+from tenacity import retry, stop_after_attempt, wait_exponential
+
+from ethereum_test_tools import Address
+
+
+class BaseRPC:
+    """
+    Represents a base RPC class for every RPC call used within EEST based hive simulators.
+    """
+
+    def __init__(self, client):
+        self.client = client
```

Comment:
This is of type `hive.client.Client`, I don't think this has any business in a generic RPC class :slightly_smiling_face:.

Reply:
Yeah exactly, its so we can instantiate an RPC "client" based on the hive client we are sending the RPC requests to.

I put it here originally so we can define both `eth_rpc = EthRPC(geth_client)` and `engine_rpc = EngineRPC(besu_client)`, to then use the respective rpc functions appropriately, `engine_rpc.new_payload(...)`.

Reply:
Yes, but the hive client is only used to get the `ip` address, as far as I can tell? In which case I would suggest providing the ip to the constructor instead of a hive client. Otherwise any user of this class has to instantiate a hive client first, which is a bit weird (e.g., `src/cli/gentest.py`).